File: java/leetcode_grind/Day459.java
package leetcode_grind;

public class Day459 {
    // https://leetcode.com/problems/rearrange-array-elements-by-sign/description/
    static class Solution1 {
        public int[] rearrangeArray(int[] nums) {
            var n = nums.length;
            var ans = new int[n];
            var pos = 0;
            var neg = 1;
            for (int i = 0; i < nums.length; i++) {
                if (nums[i] > 0) {
                    ans[pos] = nums[i];
                    pos += 2;
                } else {
                    ans[neg] = nums[i];
                    neg += 2;
                }
            }
            return ans;
        }
    }
}

File: src/leetcode_grind/day_459.rs
// https://leetcode.com/problems/rearrange-array-elements-by-sign/description/
pub fn rearrange_array(nums: Vec<i32>) -> Vec<i32> {
    let mut res = vec![];
    let mut p = 0;
    let mut n = 0;
    while res.len() != nums.len() {
        while nums[p] < 0 {
            p += 1;
        }
        while nums[n] > 0 {
            n += 1;
        }
        res.push(nums[p]);
        res.push(nums[n]);
        p += 1;
        n += 1;
    }
    res
}
